src/core/Database.h

```diff
@@ -188,30 +188,33 @@ public slots:
         QScopedPointer<PasswordKey> challengeResponseKey;
 
         QSharedPointer<const CompositeKey> key;
-        QSharedPointer<Kdf> kdf = QSharedPointer<AesKdf>::create(true);
+        QSharedPointer<Kdf> kdf;
 
         QVariantMap publicCustomData;
 
         DatabaseData()
-            : masterSeed(new PasswordKey())
-            , transformedDatabaseKey(new PasswordKey())
-            , challengeResponseKey(new PasswordKey())
         {
-            kdf->randomizeSeed();
+            clear();
         }
 
         void clear()
         {
+            resetKeys();
             filePath.clear();
+            publicCustomData.clear();
+        }
 
-            masterSeed.reset();
-            transformedDatabaseKey.reset();
-            challengeResponseKey.reset();
+        void resetKeys()
+        {
+            masterSeed.reset(new PasswordKey());
+            transformedDatabaseKey.reset(new PasswordKey());
+            challengeResponseKey.reset(new PasswordKey());
 
             key.reset();
-            kdf.reset();
 
-            publicCustomData.clear();
+            // Default to AES KDF, KDBX4 databases overwrite this
+            kdf.reset(new AesKdf(true));
+            kdf->randomizeSeed();
```

Comment:
Is this a good idea? Shouldn't be default to something more secure?

Reply:
I tried to fix this, but we would have to recode a bunch of places that make assumptions on kdf settings. This is specifically so for kdbx3 reader and the encryption settings widget. Note the original code set this as well, but it did so only once and did not properly reset these settings when the database object was reset.